💄 Add post page

devsoob opened the pull request:
게시글 상세 정보, 댓글 리스트, 댓글 입력 각 컴포넌트 분리함.
댓글 유무, 로그인 유무에 따라 보여줄 컴포넌트 분리함.

devsoob committed:
💄 Add post page
게시글 상세 정보, 댓글 리스트, 댓글 입력 각 컴포넌트 분리함.
댓글 유무, 로그인 유무에 따라 보여줄 컴포넌트 분리함.

synuns commented on src/components/post/AsideUser.jsx:
```diff
@@ -0,0 +1,135 @@
+      </UserAvatar>
+      <UsernameWrapper>
+        {/* <NotLoggedIn> 삭제함 */}
+        <LoggedIn>
```
코드를 복사해서 사용한 건 좋은데 컴포넌트의 속성에 맞게 직관적으로 이름을 수정하시면 좋겠습니다

devsoob replied:
LoggedIn 이름을 StPostUser 로 수정했습니다.

synuns commented on src/components/post/AsideUser.jsx:
```diff
@@ -0,0 +1,135 @@
+          <UserBadgeBox>
+            {/* 직군, 연차 정보가 부족한 경우 */}
+            {/* <UserBadge>직군, 연차 정보가 부족합니다.</UserBadge> */}
+            <UserBadge data-id="0">웹개발</UserBadge>
+            <UserBadge>신입</UserBadge>
```
작성자의 프로필이 완성되지 않은 경우의 뱃지도 만들어주셨으면 좋겠습니다.

devsoob replied:
의논 완료 - 작성자의 프로필이 완성되지 않은 경우, 게시글 작성 페이지에서 마이 페이지로 리다이렉트
(작성자의 프로필이 완성되지 않은 경우 뱃지 없음)

synuns commented on src/components/post/CommentInput.jsx:
```diff
@@ -0,0 +1,134 @@
+          <button type="submit">등록</button>
+        </form>
+      </Bottom>
+      <StBtn>
+        <Link to={`/`}>
+          <button type="button" onClick={handleSubmit}>
+            <img src={arrow} alt="arrow" />
+            목록으로
+          </button>
+        </Link>
+      </StBtn>
```
버튼 컴포넌트면 Link를 사용하지 말고 handler로 이동하도록 구현하시는게 좋습니다.
버튼은 onClick이라는 유용한 이벤트핸들러가 있으니까요

devsoob replied:
버튼 onClick 시 handler 로 이동하도록 아래와 같이 구현했습니다.

```
  const LinkToCommunity = () => {
    navigate('/');
  };
```

synuns commented on src/components/post/CommentList.jsx:
```diff
@@ -0,0 +1,51 @@
+const CommentList = () => {
+  return (
+    <>
+      <CommentItem>
+        <PostUser />
+        <CommentContent>
+          It is a long established fact that a reader will be distracted by the
+          readable content of a page when looking at its layout. The point of
+          using Lorem Ipsum is that it has a more-or-less normal distribution of
+          letters, as opposed to using 'Content here, content here', making it
+          look like readable English.
+        </CommentContent>
+      </CommentItem>
+      <CommentItem>
+        <PostUser />
+        <CommentContent>
+          It is a long established fact that a reader will be distracted by the
+          readable content of a page when looking at its layout. The point of
+          using Lorem Ipsum is that it has a more-or-less normal distribution of
+          letters, as opposed to using 'Content here, content here', making it
+          look like readable English.
+        </CommentContent>
+      </CommentItem>
```
mock data를 배열로 만들어서 api 명세에 맞게 map해주는 식으로 구현하시면 추후에 수정도 적어지고 좋겠습니다
추가로 CommentItem을 Comment로 이름 수정하시고 따로 컴포넌트로 빼시는게 훨씬 이점이 많습니다

devsoob replied:
1. mock data를 배열로 만들어서 Post.jsx (Post 페이지)에 추가한 후 아래와 같이 map 으로 구현했습니다.
```      
{comments &&
        comments.map((comment) => (
          <Comment key={comment.commentId} comment={comment} />
))}
```
2.  CommentItem을 Comment로 이름 수정하시고 따로 컴포넌트로 뺐습니다.

synuns commented on src/components/post/EmptyCommentInput.jsx:
```diff
@@ -0,0 +1,94 @@
+          </form>
+        </Link>
+      </Bottom>
+      <StBtn>
+        <Link to={`/`}>
+          <button type="button">
+            <img src={arrow} alt="arrow" />
+            목록으로
+          </button>
+        </Link>
```
버튼은 Link 말고 핸들러함수로

devsoob replied:
버튼 onClick 시 handler 로 이동하도록 아래와 같이 구현했습니다.

```
  const LinkToCommunity = () => {
    navigate('/');
  };
```

synuns commented on src/components/post/EmptyCommentList.jsx:
```diff
@@ -0,0 +1,32 @@
+import theme from '../../styles/theme';
+import empty from '../../assets/empty_comment.svg';
+
+const EmptyCommentList = () => {
+  return (
+    <StEmptyCommentList>
+      <img src={empty} alt="speech-bubble" />
+      <p>첫 댓글을 남겨주세요</p>
+    </StEmptyCommentList>
+  );
+};
```
빈 댓글 처리 컴포넌트 너무 좋습니다👍👍

synuns commented on src/components/post/MyAvatar.jsx:
```diff
@@ -0,0 +1,35 @@
+const MyAvatar = () => {
+  return (
+    <>
+      <UserAvatar>
+        <img src={defaultProfile} alt="profile" />
+      </UserAvatar>
```
userAvatar 컴포넌트는 사용할 곳이 많으니 post 폴더 말고 User나 Common폴더로 옮겨놓으면 좋을 것 같습니다

synuns replied:
해당사항은 제가 추가해놓았으니 merge후에 컴포넌트 변경하겠습니다.
pr 확인 부탁드려요

synuns commented on src/components/post/PostContent.jsx:
```diff
@@ -0,0 +1,97 @@
+
+const PostContent = () => {
+  return (
+    <StPostContent>
+      <StPostWrapper>
+        <PostUser />
+        <Content>
+          <h1>게시글 제목</h1>
+          <div className="content">
+            It is a long established fact that a reader will be distracted by
+            the readable content of a page when looking at its layout. The point
+            of using Lorem Ipsum is that it has a more-or-less normal
+            distribution of letters, as opposed to using 'Content here, content
+            here', making it look like readable English. Many desktop publishing
+            packages and web page editors now use Lorem Ipsum as their default
+            model text, and a search for 'lorem ipsum' will uncover many web
+            sites still in their infancy. Various versions have evolved over the
+            years, sometimes by accident, sometimes on purpose (injected humour
+            and the like).
+          </div>
+          <PostTags>선택한 태그들</PostTags>
+          <PostAction />
+        </Content>
+      </StPostWrapper>
+      <StComment>
+        {/* 댓글이 없는 경우 */}
+        {/* <EmptyCommentList /> */}
+        {/* 댓글 있는 경우 */}
+        <CommentList />
+        {/* 로그인 안된 경우 */}
+        {/* <EmptyCommentInput /> */}
+        {/* 로그인 된 경우 */}
+        <CommentInput />
+      </StComment>
+    </StPostContent>
```
mock data를 상단에 선언해주시고 해당 data를 return 해주는 식으로 작성해주시기 바랍니다.
props를 전달해야하는 상황이면 그에 맞게 대응해서 작성해주세요

devsoob replied:
mock data를 배열로 만들어서 Post.jsx (Post 페이지)에 추가한 후 props를 PostContent 컴포넌트에 전달했습니다.

synuns reviewed (commented):
리뷰 남겼으니 읽어보시고 수정 부탁드립니다.
커밋은 잘 나눠서 더 자주하셨으면 좋겠습니다

devsoob committed:
💄 Update aside bar UI
post 페이지의 aside bar 컴포넌트 aside 폴더로 분리

devsoob committed:
💄 Update comment UI
post 페이지의 댓글 관련 컴포넌트 comment 폴더로 분리 및 수정

devsoob committed:
💄 Update post UI
post 페이지의 게시글 관련 컴포넌트 postDetail 폴더로 분리 및 수정,
- 파일명 변경 : PostAction -> BottomAction
- 파일명 변경 및 이동: PostContent -> PostLayout 이름 변경 후 상위 폴더인 post 이동

devsoob committed:
💄 Update post page
- PostLayout : post page 의 컴포넌트들 전제 구조
- Post page : 임시 mock data 작업 후 하위 컴포넌트에 props 전달함
- MyAvatar : 로그인한 유저의 사진 및 정보 컴포넌트(대체 필요)

devsoob committed:
🍱 Add assets
post page 게시글 공유 버튼과 편집 버튼(수정 및 삭제) 이미지 추가

devsoob commented:
post 페이지 수정 필요
- [ ] 게시글 좋아요 클릭 시 아이콘 파란색으로 바꾸고 count 숫자 추가(Action과 BottomAction 컴포넌트 동시 적용 필요)
- [ ] 댓글 작성 시 아이콘 파란색으로 바꾸고 count 숫자 추가(Action과 BottomAction 컴포넌트 동시 적용 필요)
- [ ] 편집 버튼 클릭 시 삭제하기 & 수정하기 작은 모달 띄우고 각각 기능 넣기(EditPost 컴포넌트)
- [ ] 로그인한 유저의 프로필 이미지와 닉네임 보여주기(commentInput 컴포넌트)
- [ ] 댓글 삭제 UI 및 기능 추가
